ledger/ledger-on-sql/src/main/scala/com/daml/ledger/on/sql/SqlLedgerReaderWriter.scala

```diff
@@ -53,30 +47,22 @@ final class SqlLedgerReaderWriter(
     participantId,
     () => timeProvider.getCurrentTime,
     SubmissionValidator.create(SqlLedgerStateAccess),
-    latestSequenceNo => dispatcher.signalNewHead(latestSequenceNo + 1),
+    latestSequenceNo => dispatcher.signalNewHead(latestSequenceNo),
   )
 
   override def currentHealth(): HealthStatus = Healthy
 
-  override def events(offset: Option[Offset]): Source[LedgerEntry, NotUsed] =
+  override def events(startExclusive: Option[Offset]): Source[LedgerEntry, NotUsed] =
     dispatcher
       .startingAt(
-        offset.getOrElse(StartOffset).components.head,
+        KVOffset.highestIndex(startExclusive.getOrElse(StartOffset)),
         RangeSource((start, end) => {
           Source
-            .futureSource(database
-              .inReadTransaction(s"Querying events [$start, $end[ from log") { queries =>
+            .future(database
```

Comment:
Why not use `.futureSource` and wrap `queries.selectFromLog(start, end)` in a `Source` with `.map(Source _)`?

Reply:
My mental mostly is mostly around initializing the stream and afterwards manipulating it in the way it's necessary. In this case it doesn't matter much I guess. Other than we then probably should use `DirectExecutionContext` to avoid having a thread context switch for simply turning `Seq[_]` to `Source[_]`.

Reply:
Eh, leave it for now, and I'll see if I can get rid of the futures entirely at a later point. They're kind of pointless.